File: src/main/java/hu/elte/alkfejl/issuetracker/model/Issue.java
package hu.elte.alkfejl.issuetracker.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.sql.Timestamp;
import java.time.LocalDateTime;
import java.util.List;

/**
 * @author Godzsák Dávid <godzsakdavid@gmail.com>
 */
@Entity
@Table(name = "ISSUES")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Issue extends BaseEntity {

    @JoinColumn
    @ManyToOne(targetEntity = User.class)
    private User user;

    @Column(nullable = false)
    private Timestamp timestamp;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    @Column(nullable = false)
    private String location;

    @Column(nullable = false)
    private String description;

    @JoinColumn
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, targetEntity = IssueMessage.class)
    private List<IssueMessage> messages;

    public enum Status {
        ADDED, READY, ONGOING
    }
}

File: issue-tracker/src/model/Issue.ts
import {User} from "./User";

export enum IssueStatus {
  ADDED, READY, ONGOING
}

export class Issue {
  user: User;
  timeStamp: Date;
  status: IssueStatus;
  description: String;
  location: String;

  constructor(description?: String, location?: String, status?: IssueStatus, timeStamp?: Date, user?: User) {
    this.user = user;
    this.timeStamp = timeStamp;
    this.status = status;
    this.description = description;
    this.location = location;
  }
}
